x/batch/msg_test.go

```diff
@@ -1,42 +1,53 @@
 package batch_test
 
 import (
-	"errors"
 	"testing"
 
 	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/weavetest/assert"
 	"github.com/iov-one/weave/x/batch"
-	. "github.com/smartystreets/goconvey/convey"
-	"github.com/stretchr/testify/mock"
 )
 
-func TestMsg(t *testing.T) {
-	Convey("Test Validate", t, func() {
-		msg := &mockMsg{}
-		Convey("Test happy flow", func() {
-			msg.On("MsgList").Return(make([]weave.Msg, 10), nil)
-			So(batch.Validate(msg), ShouldBeNil)
-			msg.AssertExpectations(t)
-		})
+func TestMsgValidate(t *testing.T) {
+	specs := map[string]struct {
+		msg batch.Msg
+		err *errors.Error
+	}{
+		"Happy flow": {
+			msg: &mockMsg{list: make([]weave.Msg, 10)},
+		},
+		"Test list too long": {
+			msg: &mockMsg{list: make([]weave.Msg, batch.MaxBatchMessages+1)},
+			err: errors.ErrInput,
+		},
+		"Test error": {
+			msg: &mockMsg{list: make([]weave.Msg, batch.MaxBatchMessages), listErr: errors.ErrState},
+			err: errors.ErrState,
+		},
+	}
 
-		Convey("Test list too long", func() {
-			msg.On("MsgList").Return(make([]weave.Msg, 11), nil)
-			So(batch.Validate(msg), ShouldNotBeNil)
-			msg.AssertExpectations(t)
+	for msg, spec := range specs {
+		t.Run(msg, func(t *testing.T) {
+			err := batch.Validate(spec.msg)
+			if spec.err == nil {
```

Comment:
There is no need for this check. `errors.Error.Is` compares to `nil` value as well.
https://github.com/iov-one/weave/blob/master/errors/errors.go#L167-L169